src/buffer.rs

```diff
@@ -146,11 +183,20 @@ impl Buffer {
         })
     }
 
+    pub fn id(&self) -> bindings::VABufferID {
+        self.id
+    }
+
     /// Convenience function to return a `VABufferID` vector from a slice of `Buffer`s in order to
     /// easily interface with the C API where a buffer array might be needed.
     pub fn as_id_vec(buffers: &[Self]) -> Vec<bindings::VABufferID> {
         buffers.iter().map(|buffer| buffer.id).collect()
     }
+
+    /// Map VAEncCodedBufferType buffer to `CodedBuffer`
```

Comment:
This actually maps any `Buffer` to `CodedBuffer`, which is an issue. I go into more detail in another comment below.